cmd/tsbs_generate_queries/databases/mongo/devops-naive.go

```diff
@@ -42,43 +44,46 @@ func (d *NaiveDevops) GroupByTime(qi query.Query, nHosts, numMetrics int, timeRa
 	metrics, err := devops.GetCPUMetricsSlice(numMetrics)
 	panicIfErr(err)
 
-	bucketNano := time.Minute.Nanoseconds()
 	pipelineQuery := []bson.M{
 		{
-			"$match": map[string]interface{}{
+			"$match": bson.M{
 				"measurement": "cpu",
-				"timestamp_ns": map[string]interface{}{
-					"$gte": interval.StartUnixNano(),
-					"$lt":  interval.EndUnixNano(),
+				"time": bson.M{
+					"$gte": interval.Start(),
+					"$lt":  interval.End(),
 				},
-				"tags.hostname": map[string]interface{}{
+				"tags.hostname": bson.M{
 					"$in": hostnames,
 				},
 			},
 		},
 		{
-			"$project": map[string]interface{}{
+			"$project": bson.M{
 				"_id": 0,
-				"time_bucket": map[string]interface{}{
-					"$subtract": []interface{}{
-						"$timestamp_ns",
-						map[string]interface{}{"$mod": []interface{}{"$timestamp_ns", bucketNano}},
+				"time_bucket": bson.M{
+					"$dateSubtract": bson.M{
+						"startDate": "$time",
+						"unit":      "second",
+						"amount":    bson.M{"$second": bson.M{"date": "$time"}},
 					},
 				},
-
-				"fields": 1,
 			},
 		},
 	}
 
+	projectMap := pipelineQuery[1]["$project"].(bson.M)
+	for _, metric := range metrics {
+		projectMap[metric] = 1
+	}
+
 	group := bson.M{
 		"$group": bson.M{
 			"_id": "$time_bucket",
 		},
 	}
 	resultMap := group["$group"].(bson.M)
 	for _, metric := range metrics {
-		resultMap["max_"+metric] = bson.M{"$max": "$fields." + metric}
+		resultMap["max_"+metric] = bson.M{"$max": "$" + metric}
```

Comment:
Previously all the metrics were nested under a top-level field "fields", which wouldn't have worked well with time-series collections because $_internalUnpackBucket's include / exclude list only works with top-level fields and we'd have to unpack all the regular fields that were nested under "fields". Greg changed this format in a previous commit.